src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -201,17 +193,25 @@ public virtual void ApplyResult(CorsResult result, HttpResponse response)
 
             if (result.AllowedMethods.Count > 0)
             {
-                // Filter out simple methods
-                var nonSimpleAllowMethods = result.AllowedMethods
-                    .Where(m =>
-                        !CorsConstants.SimpleMethods.Contains(m, StringComparer.OrdinalIgnoreCase))
-                    .ToArray();
+                string[] allowMethods;
+                if (result.FilterSimpleMethods)
```

Comment:
Could this be done in https://github.com/aspnet/CORS/pull/174/files#diff-a347a5f04243b17ec20cf0896f393f68R159 directly? That way we would save the public API bit and the fix would be more straight forward

Reply:
AddResult gets called after the EvaluatePreflight removing these cors 'Simple Requests'. So unless a rewrite doing this check in both the EvaluateRequest aswell as the EvaluatePreflightRequest.

As far as I know there's no need for the filtering of Simple Methods outside of Preflight requests (But I'm not sure, which is why I left it in the general AddResult() function using this patch.

Reply:
We were discussing possibly not doing the filtering at all. Like you said, the spec doesn't require it. I have some reservations about doing this by default (with no "flag" to opt out 2.1). Let me get back to you by early next week once we've had a chance to discuss this.

Reply:
I was thinking about that too, Filtering is just adding extra overhead and the client is supposed to de termine if it needs to do a preflight request or not. I'll be waiting for a reply and I'm happy to remove these filters for and push them to this issue.

Reply:
We had a chat and landed on removing the filtering in entirety. The change here should be pretty-trivial - if you start afresh from release/2.2:

* Remove the filtering for methods, headers, and exposed headers
* Remove `CorsConstants.Simple*`

Does this sound like something you'd like to continue working on?